hypothesis/lms
# Youtube picker (JSTORPicker-based)

> This PR is part of https://github.com/hypothesis/lms/issues/5328

This PR adds a new content picker for YouTube videos, which is based in `JSTORPicker`.

![image](https://user-images.githubusercontent.com/2719332/235932213-718f5fa1-c518-4e21-bd1e-fc057d21d278.png)

![image](https://user-images.githubusercontent.com/2719332/236179439-475a676b-9b55-4c8c-8445-67a0617016b0.png)

![image](https://user-images.githubusercontent.com/2719332/236179378-0e595a66-5fc7-4b02-a7fb-fb178b6b03dc.png)

### Out of the scope of this PR

- Design of the content selectors. This just adds a new button as a simple entry point until the whole section is redesigned.
- Implement the logic after a valid YouTube URL has been set. That will come on a follow-up PR.

### Testing

- Check out this branch.
- Enable the feature flag by going to `http://localhost:8001/admin/instance/{instance_id}/` and checking **YouTube enabled**:
  ![image](https://user-images.githubusercontent.com/2719332/236158855-d6f339a7-640c-4f3f-ba80-3e1fd404eba6.png)
- Open an assignment pointing to your localhost lms instance
- Click on `Select video from YouTube`.
- Verify that an error is displayed if the URL is not any of the known YouTube video patterns.
- Verify a dummy thumbnail, title and channel name is displayed when a valid YouTube video is set.
- The **Continue** button does nothing for now.

### TODO

- [x] Add a dummy video title and channel name to `URLFormWithPreview`
- [x] Add tests for `URLFormWithPreview`
- [x] Add tests to `YouTubePicker`
